src/EditorFeatures/CSharpTest/MakeFieldReadonly/MakeFieldReadonlyTests.cs

```diff
@@ -0,0 +1,616 @@
+        public async Task FieldIsPublic()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        public int [|_foo|];
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldIsInternal()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        internal int [|_foo|];
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldIsEvent()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private event System.EventHandler [|Foo|];
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldIsReadonly()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private readonly int [|_foo|];
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldNotAssigned()
+        {
+            await TestInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|];
+    }
+}",
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private readonly int _foo;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldNotAssigned_Struct()
+        {
+            await TestInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    struct MyStruct
+    {
+        private int [|_foo|];
+    }
+}",
+@"namespace ConsoleApplication1
+{
+    struct MyStruct
+    {
+        private readonly int _foo;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldAssignedInline()
+        {
+            await TestInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|] = 0;
+    }
+}",
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private readonly int _foo = 0;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task MultipleFieldsAssignedInline_AllCanBeReadonly()
+        {
+            await TestInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|] = 0, _bar = 0;
+    }
+}",
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int _bar = 0;
+        private readonly int _foo = 0;
```

Comment:
OK. I split them to make the fix simpler, and because I don't like fields being declared together anyway. It should be fairly trivial to move some of the code to a helper file and use it in the code fix too.